enhancements/machine-api/converting-machine-api-to-cluster-api.md

```diff
@@ -0,0 +1,756 @@
+This is possible even with the current proposal as an additional feature which may be desirable for some users to create additional MachineSets, based on existing MachineSets.
+
+However, this doesn't help with the goal of removing Machine API controllers in the longer term, so is not sufficient to meet our goals on its own.
+
+### Deletion events are always synchronised 
+
+In the scenario described above, users may delete Machine API resources without impacting the Cluster API resource providing the Cluster API resource is authoritative.
+This means that the user may clean up old Machine API resources without any special consideration.
+
+This eventually will allow the Machine API resources to be deleted and have no effect on the running system.
+
+Instead, we could always synchronise deletion events.
+This would mean that any delete of a resource is always mirrored no matter the circumstances.
+
+This would mean that, to allow users to clean up Machine API resources, we would need to introduce some “escape hatch”,
+likely an annotation, that a user could set on the resource to allow it to be deleted without also deleting the Cluster API equivalent.
+
+This could lead to users believing that, because the Cluster API resource is authoritative, deleting the Machine API resource is acceptable.
+Instead, it would result in deleting a Machine which they had not intended to do.
+This could be catastrophic to clusters if users accidentally delete a control plane Machine,
+therefore we need to make this as simple as possible, by having the single authoritative switch indicate the behaviour of deletion,
+there is less room for error on the customers part.
+
+### Use webhooks to synchronise annotations
+
+Instead of having each controller look up the annotation on the Machine API resource,
+we could use webhooks on both Machine API and Cluster API resources to ensure that only one of the resources is authoritative at any time.
+
+With this approach, the authoritative annotation existing on a resource would indicate that it is the authoritative version, and therefore should be reconciled. 
+Webhooks would be in place to prevent the annotation from being added to both Machine API and Cluster API mirrors at the same time.
+
+To move between the two versions, the user would first have to remove the annotation from one (in which case no reconciliation happens anywhere), and then add it to the other to make it authoritative.
+
+This approach may seem to be less complex, but has the potential to cause issues if, for some reason, the webhooks do not work correctly (eg a change is made while the webhook is being restarted).
+
+To avoid this potential, our approach suggests to have the authoritative annotation exist in only one location, on the Machine API resource.
+
+### Keep the authoritative annotation on the Cluster API resource
```

Comment:
is this part of the "Alternatives" section?

Reply:
Yes, it's an alternative to keeping the annotation on the MAPI resource, though that's not so obvious right now